k8s/core/configmap/configmap_lock_v2.go

```diff
@@ -174,13 +176,19 @@ func (c *configMap) IsKeyLocked(key string) (bool, string, error) {
 	}
 
 	if owner, ok := lockIDs[key]; ok {
-		// Someone owns the lock: let's check the expiration and the owner
+		// Existing key is unlocked if
+		//   1. Lock has expired; or
+		//   2. Lock owned by itself but refresh goroutine is not running
 		expiration := lockExpirations[key]
 		if time.Now().After(expiration) {
-			// Lock is expired
 			return false, "", nil
 		}
-
+		c.kLocksV2Mutex.Lock()
+		defer c.kLocksV2Mutex.Unlock()
```

Comment:
- It's not necessary defer the unlock which will hold the lock until the end of the funciton. You can unlock after line 188
- In the refreshLock function, you are using lock.Lock() to update refreshing. It'll be more appropriate to use the samelock for access in line 189. That unlock you could defer

Reply:
I'm not understanding the second point. How to use the same lock in line 189 in IsKeyLocked() in refreshLock()?

Reply:
Clarified offline